ext/storage_sources/gcp_store/test/test_gcp_connection.cpp

```diff
@@ -30,5 +30,338 @@
+
+    return (0);
+}
+
+TEST_CASE("Testing class gcpConnection", "gcp-connection")
+{
+    // Setup the test environment.
+    REQUIRE(setup_test_defaults() == 0);
+
+    gcp_connection conn(test_defaults::bucket_name, test_defaults::obj_prefix);
+    // bool exists = false;
+    // size_t objectSize;
+
+    const std::string object_key = "test_object";
+    const std::string file_name = "test_object.txt";
+    const std::string file_path = "./" + file_name;
+    const std::string non_exi_object_key = "test_non_exist";
+    const std::string non_exi_file_name = "test_non_exist.txt";
+    const bool list_single = true;
+    std::vector<std::string> objects;
+
+    google::cloud::storage::Client client = conn.get_client();
+
+    std::ofstream File(file_name);
+    std::string payload = "Test payload :)";
+    File << payload;
+    File.close();
+
+    SECTION("Simple list test", "[gcp-connection]")
+    {
+        const int32_t total_objects = 20;
+
+        // No matching objects. Object size should be 0.
+        REQUIRE(conn.list_objects(objects, false) == 0);
+        REQUIRE(objects.size() == 0);
+
+        // No matching objects with list_single. Object size should be 0.
+        REQUIRE(conn.list_objects(objects, list_single) == 0);
+        REQUIRE(objects.size() == 0);
+
+        // Upload 1 file and test list.
+        objects.clear();
+        client.UploadFile(
+          file_path, test_defaults::bucket_name, test_defaults::obj_prefix + object_key);
+        REQUIRE(conn.list_objects(objects, false) == 0);
+        REQUIRE(objects.size() == 1);
+
+        // Delete that object and test list.
+        objects.clear();
+        client.DeleteObject(test_defaults::bucket_name, test_defaults::obj_prefix + object_key);
+        REQUIRE(conn.list_objects(objects, false) == 0);
+        REQUIRE(objects.size() == 0);
+
+        // Upload multiple files and test list.
+        objects.clear();
+        for (int i = 0; i < total_objects; i++) {
+            client.UploadFile(file_path, test_defaults::bucket_name,
+              test_defaults::obj_prefix + std::to_string(i) + ".txt");
+        }
+        REQUIRE(conn.list_objects(objects, false) == 0);
+        REQUIRE(objects.size() == total_objects);
+
+        // Test if list single only returns one object.
+        objects.clear();
+        REQUIRE(conn.list_objects(objects, list_single) == 0);
+        REQUIRE(objects.size() == 1);
+
+        for (int i = 0; i < total_objects; i++) {
+            client.DeleteObject(
+              test_defaults::bucket_name, test_defaults::obj_prefix + std::to_string(i) + ".txt");
+        }
+    }
+
+    SECTION("Simple put test", "[gcp-connection]")
+    {
+
+        // Upload a file that does not exist locally - should fail.
+        REQUIRE(conn.put_object(non_exi_object_key, non_exi_file_name) == ENOENT);
+
+        // Check number of files with the given prefix currently in bucket
```

Comment:
Please make sure all your comments are full sentences and have proper punctuation.

Reply:
done